include/deal.II/fe/fe_interface_values.h

```diff
@@ -0,0 +1,608 @@
+   * number of DoFs on the two adjacent cells.
+   */
+  unsigned
+  n_interface_dofs() const;
+
+  /**
+   * Return if the current interface, which is set by reinit(), is an internal
+   * face with two adjacent cells or a boundary face.
+   */
+  bool
+  at_boundary() const;
+
+  /**
+   * Return the set of joint DoF indices. This includes indices from both cells.
+   */
+  std::vector<types::global_dof_index>
+  get_interface_dof_indices() const;
+
+  /**
+   * Translate a local "interface dof index" to a local "dof index" of the
+   * underlying FEFaceValues.
+   */
+  unsigned int
+  interface_dof_index_to_fe_dof_index(unsigned int interface_dof_index) const;
+
+  /**
+   * For a given "interface dof index" return whether it belongs to the cell (0)
+   * or to the neighbor (1).
+   */
+  unsigned int
+  interface_dof_index_to_fe_index(const unsigned int interface_dof_index) const;
+
+  /**
+   * Convert a FiniteElement index @p fe_index (0=cell, 1=neighbor) and dof
+   * index @p face_dof_index into an interface DoF index.
+   *
+   * The inverse of this operation is done with
+   * interface_dof_index_to_fe_index() and
+   * interface_dof_index_to_fe_dof_index().
+   */
+  unsigned int
+  interface_dof_index(const unsigned int fe_index,
+                      const unsigned int face_dof_index) const;
+
+  /**
+   * Return the FEFaceValue object of the cell or the neighboring cell
+   *
+   * If @p cell_or_neighbor is 0, return the cell's, if it is 1, return the
+   * neighboring cell's FEFaceValue object.
+   *
+   * @note The argument @p cell_or_neighbor is returned by
+   * interface_dof_index_to_fe_index().
+   */
+  const FEFaceValuesBase<dim, spacedim> &
+  get_fe_values(const unsigned int cell_or_neighbor) const;
+
+  /**
+   * Return the FEFaceValue object of the current cell
+   */
+  const FEFaceValuesBase<dim, spacedim> &
+  get_fe_values() const;
+
+  /**
+   * Return the FEFaceValue object of the current neighboring cell
+   */
+  const FEFaceValuesBase<dim, spacedim> &
+  get_fe_values_neighbor() const;
+
+  /**
+   * Return the normal in a given quadrature point.
+   */
+  Tensor<1, spacedim>
+  normal(const unsigned int q_point_index) const;
+
+  /**
+   * Return the current (if @p current_cell is true) or neighboring (if @p
+   * current_cell is false) value on the interface for the shape function @p
+   * interface_dof_index in the quadrature point @p q_point of component @p
+   * component.
+   *
+   * @note This function is typically used to pick the upstream or downstream
+   * value based on a direction. This can be achieved by using
+   * <code>(direction * normal)>0</code> as the first argument of this
+   * function.
+   */
+  double
+  choose(const bool         current_cell,
+         const unsigned int interface_dof_index,
+         const unsigned int q_point,
+         const unsigned int component = 0) const;
+
+  /**
+   * Return the jump $[u]=u_1 - u_2$ on the interface for the shape function
```

Comment:
I don't think $1$ and $2$ have any meaning in this context. Maybe use `cell` and `neighbor`?
```suggestion
   * Return the jump $[u]=u_{\text{cell}} - u_{\text{neighbor}}$ on the interface for the shape function
```